tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_section_order.py

```diff
@@ -0,0 +1,220 @@
+"""
+Different variations of EOF sections displacement
+"""
+
+from enum import Enum
+from typing import List
+
+import pytest
+
+from ethereum_test_tools import EOFTestFiller
+from ethereum_test_tools import Opcodes as Op
+from ethereum_test_tools.eof.v1 import (
+    AutoSection,
+    Bytes,
+    Container,
+    EOFException,
+    Section,
+    SectionKind,
+)
+from ethereum_test_tools.eof.v1.constants import NON_RETURNING_SECTION
+
+from .. import EOF_FORK_NAME
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-3540.md"
+REFERENCE_SPEC_VERSION = "8dcb0a8c1c0102c87224308028632cc986a61183"
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+class SectionTest(Enum):
+    """
+    Enum for the test type
+    """
+
+    MISSING = 1
+    WRONG_ORDER = 2
+
+
+class CasePosition(Enum):
+    """
+    Enum for the test position
+    """
+
+    BODY = 1
+    HEADER = 2
+    BODY_AND_HEADER = 3
+
+
+def get_expected_code_exception(
+    section_kind, section_test, test_position
+) -> tuple[str, EOFException | None]:
+    """
+    Verification vectors with code and exception based on test combinations
+    """
+    match (section_kind, section_test, test_position):
+        case (SectionKind.TYPE, SectionTest.MISSING, CasePosition.HEADER):
+            return "ef000102000100030400010000800001305000ef", EOFException.MISSING_TYPE_HEADER
+        case (SectionKind.TYPE, SectionTest.MISSING, CasePosition.BODY):
+            return (
+                "ef0001010004020001000304000100305000ef",
+                EOFException.INVALID_SECTION_BODIES_SIZE,
+            )
+        case (SectionKind.TYPE, SectionTest.MISSING, CasePosition.BODY_AND_HEADER):
+            return "ef0001020001000304000100305000ef", EOFException.MISSING_TYPE_HEADER
+        case (SectionKind.TYPE, SectionTest.WRONG_ORDER, CasePosition.HEADER):
+            return (
+                "ef000102000100030100040400010000800001305000ef",
+                EOFException.MISSING_TYPE_HEADER,
+            )
+        case (SectionKind.TYPE, SectionTest.WRONG_ORDER, CasePosition.BODY):
+            return (
+                "ef000101000402000100030400010030500000800001ef",
+                # TODO why invalid first section type? it should say that the body incorrect
```

Comment:
I mean I am confused because I separate header and body. so first section in the header is correct but in the body are wrong bytes. would been nice to have this marked in exception.